includes/admin/class-sensei-onboarding-api.php

```diff
@@ -0,0 +1,224 @@
+<?php
+/**
+ * Onboarding REST API.
+ *
+ * @package Sensei\Onboarding
+ * @since   3.1.0
+ */
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly.
+}
+
+/**
+ * Sensei Onboarding REST API endpoints.
+ *
+ * @package Sensei
+ * @author  Automattic
+ * @since   3.1.0
+ */
+class Sensei_Onboarding_API {
+
+	/**
+	 * Main Onboarding instance.
+	 *
+	 * @var Sensei_Onboarding
+	 */
+	private $onboarding;
+
+	const PURPOSES = [ 'share_knowledge', 'generate_income', 'promote_business', 'provide_certification', 'train_employees', 'other' ];
+
+	/**
+	 * Sensei_Onboarding_API constructor.
+	 *
+	 * @param Sensei_Onboarding $onboarding Onboarding instance.
+	 */
+	public function __construct( $onboarding ) {
+		$this->onboarding = $onboarding;
+	}
+
+	/**
+	 * Register the REST API endpoints for Onboarding.
+	 */
+	public function register() {
+		$common = [
+			'endpoint' => [
+				'permission_callback' => [ $this, 'can_user_access_rest_api' ],
+			],
+			'arg'      => [
+				'validate_callback' => 'rest_validate_request_arg',
+			],
+		];
+
+		$welcome_endpoint = [
+			[
+				'methods'  => 'GET',
+				'callback' => [ $this, 'welcome_get' ],
+			],
+			[
+				'methods'  => 'POST',
```

Comment:
... and `WP_REST_Server::CREATABLE` or `WP_REST_Server::EDITABLE` here.

Reply:
Also 200557bc403a4a44ae4e24150a09e2a3b7d215b9. `EDITABLE` made more semantic sense as it's transforming the state of the wizard.